docs/sphinx/user_guide/feature/plugins.rst

```diff
@@ -0,0 +1,129 @@
+
+* ``RAJA::util::init_plugins();`` - Will call the ``init`` function of every currently loaded plugin.
+
+* ``RAJA::util::init_plugins("path/to/plugins");`` - Does the same as the above call to init_plugins, but will also dynamically load plugins located at the path specified.
+
+* ``RAJA::util::finalize_plugins();`` - Will call the ``finalize`` function of every currently loaded plugin. 
+
+
+------------
+Creating Plugins For RAJA
+------------
+
+Plugins take advantage of *polymorphism*, using ``RAJA::util::PluginStrategy`` as the parent and implementing the required functions for the API. An example implementation can be found at the bottom of this page.
+
+^^^^^^^^^^^
+Functions
+^^^^^^^^^^^
+The preLaunch and postLaunch functions are automatically called by RAJA before and after loop execution. This applies to RAJA's kernel and forall implementations.
+
+* ``void init(PluginOptions p) {}`` - runs on all plugins when the user makes a call to ``init_plugins``
+
+* ``void preCapture(PluginContext p) {}`` - Will occur before capture of kernel/forall.
+
+* ``void postCapture(PluginContext p) {}`` - Will occur after capture of kernel/forall.
+
+* ``void preLaunch(PluginContext& p) {}`` - Will occur before kernel/forall execution.
+
+* ``void postLaunch(PluginContext& p) {}`` - Will occur after kernel/forall execution.
+
+* ``void finalize() {}`` - runs on all plugins when the user makes a call to ``finalize_plugins``
+
+Init and finalize are never run by RAJA by default and are only run when the user makes a call to RAJA::util::init_plugin() or RAJA::util::finalize_plugin() respectively.
+
+^^^^^^^^^^^^^^^^^
+Static Loading
+^^^^^^^^^^^^^^^^^
+If the plugin is to be loaded into a project at compile time, adding the following one-liner will add the plugin to the RAJA PluginRegistry and will be loaded every time the compiled executable is run. This requires the plugin to be loaded with either in an ``#include`` statement within the project or by compiler commands.
+::
+
+  static RAJA::util::PluginRegistry::add<PluginName> P("Name", "Description");
+
+
+^^^^^^^^^^^^^^^^^
+Dynamic Loading
+^^^^^^^^^^^^^^^^^
+If the plugin is to be dynamically loaded to a project during runtime, the RAJA Plugin API requires a few conditions to be met. The following must be true about the plugin, not necessarily of the project using it.
+
+1. **The plugin must have following factory function.** This will return a pointer to an instance of your plugin. Thanks to the ``extern "C"`` a project will be able to search for "getPlugin" within the dynamically loaded plugin correctly.
```

Comment:
missing the

Reply:
I'm not sure I see any missing the's, where exactly?

Reply:
The plugin must have the following factory function